### ISSUE TITLE
Promote "stable" betas to nuget.org

### ISSUE THREAD

**Issue by christophwille** [Azure/azuremonitor-opentelemetry-profiler-net#11]
as it would make testing on larger (internal) solutions a lot easier (PATs in checked-in solutions aren't the nicest thing)

**Comment by christophwille** [comment]
We build containers on the CI servers.

**Comment by xiaomi7732** [comment]
@christophwille We are looking into it.

Can you confirm:
 
* Is your CI wired up with test environment or production environment?
* What is your CI based on? Is it GitHub action? Azure DevOps? Other?

There are some ways to keep the PAT secure while using CI, depends on specific CI pipelines, for example, with this: https://blog.devops.dev/consuming-private-nuget-feeds-from-a-dockerfile-in-a-secure-and-devops-friendly-manner-b5c90ea90bba. Is that something that you are willing to implement?

We are also looking into other ways, hopefully to make it easier.

**Comment by christophwille** [comment]
The CI only produces container images (using dotnet publish), and those are picked up by CD (to any env, even ephemeral), and in the case where I'd like to give it a try is Azure DevOps.

**Comment by xiaomi7732** [comment]
Hey @christophwille, thanks for the info. Now I see, let me repeat so that I understand it, your CI creates containers, and your CD decides the environment to deliver/deploy the containers to.

Just to double check, that I read you correct that your CI and CD pipelines (that you are trying this with) are all using Azure DevOps, right?

**Comment by christophwille** [comment]
Yes, it is all ADO (couldn't convince myself yet that it is big enough for Argo CD)

**Comment by xiaomi7732** [comment]
@christophwille in that case, do you have your own artifacts feed? If we upload the NuGet packages, can you download them and host the packages in your own feed? Will that make it easier?

**Comment by christophwille** [comment]
That would be an option

**Comment by j-zhong-ms** [comment]
Hi @christophwille, thank you for working with us on this!

We have a few temporary workarounds that do not require a PAT, and our goal remains eventually pushing the packages to official the nuget feed.

Option 1 - You can restore the packages from our temporary self-hosted feed. You won't need 
a token to access the package. [Instructions here](https://github.com/Azure/azuremonitor-opentelemetry-profiler-net/blob/main/docs/UseBagetter.md). 

Option 2 - You can update your project to reference the assembly files directly by following [these steps](https://github.com/Azure/azuremonitor-opentelemetry-profiler-net/blob/main/docs/UseAssembly.md). This option would likely take more time than option 1.

Option 3 - Continuing the previous thread, you can host the nupkg in your ADO feed [Publish and restore NuGet packages with the NuGet CLI - Azure Artifacts | Microsoft Learn](https://learn.microsoft.com/en-us/azure/devops/artifacts/nuget/publish?view=azure-devops)

Please let us know if any of these options would work for you. Again, really appreciate your feedback!

**Comment by xiaomi7732** [comment]
@christophwille Happy new year. Do you mind you take a look at the options provided by @j-zhong-ms, and see which one fits for you the best? Looking forward to hearing from you.

**Comment by xiaomi7732** [comment]
@christophwille Thanks for the feedback. Can you check this out (No.1): http://172.179.151.190:5000/

![Image](https://github.com/user-attachments/assets/04557cbf-d54c-4232-97d6-1126001bc5c1)

You could either use it directly as a NuGet source or download the packages. It's our early access feed.

**Comment by christophwille** [comment]
So in general adding to the ADO feed works with that approach (dl from Bagetter, upload to ADO feed).

**Comment by xiaomi7732** [comment]
@christophwille Thanks for the feedback. We are working on getting the package onto `nuget.org`. At the same time, please let us know if you see any issue.

**Comment by christophwille** [comment]
If anyone needs a sample with bagetter configured (and validated it works), please see https://github.com/christophwille/OTel-de-Wille/tree/main/azuremonitor-profiler
